### ISSUE TITLE
Last OS X dependancy

### ISSUE THREAD

**Issue by evi-snowm** [matiassingers/ipa-metadata#9]
Hi again,

I've been having a quick look at the current codebase with the intent of removing the last dependency on OS X only functionality. As far as I can tell, there is one left and it's in https://github.com/matiassingers/entitlements, which has an exec command to the OS X only command 'codesign'.

I can think of two places to handle the situation that the code is run on another OS, either here in ipa-metadata or in entitlements itself. I'm imaging a simple check like e.g. require('os').platform() === 'darwin' or something along those lines, since the alternative is either doing the call and hoping apple never changes their error codes or depending on a module like which.

The code change would be easiest to do in entitlements (simply returning an empty JSON if the OS isn't OS X), but it might be neater to handle in in ipa-metadata and keep entitlements throwing errors if it can't work. I would strongly prefer avoiding duplicate code in ipa-metadata though (giant if round the async, so sad you can't make parts optional in that) and I'm not really happy with what I've come up with so far to do that. Could I get your opinion on this, please?

Snippet
```javascript
unzipper.on('extract', function() {
    var path = glob.sync(output.path + '/Payload/*/')[0];
    data.metadata = plist.readFileSync(path + 'Info.plist');
    var callback = function(error, results){
      if(error) {
        console.log(error);
        return cleanUp(error);
      }

      if(Array.isArray(results)) { //brrr
        data.provisioning = results[0];
        data.entitlements = results[1];
      } else {
        data.provisioning = results;
      }
      delete data.provisioning.DeveloperCertificates;
      return cleanUp();
    };

    if(require('os').platform() === 'darwin') {
      async.parallel([
        async.apply(provisioning, path + 'embedded.mobileprovision'),
        async.apply(entitlements, path)
      ], callback);
    } else {
      provisioning(path + 'embedded.mobileprovision', callback);
    }
  });
```

**Comment by matiassingers** [comment]
Very good point @evi-snowm!

Thanks for looking into all these cross platform issues.

Take a look at what I did in the [`feature/better-cross-platform-support`](https://github.com/matiassingers/ipa-metadata/tree/feature/better-cross-platform-support branch)

----

NB. I should write some unit tests for this part at some point

**Comment by evi-snowm** [comment]
You're welcome @matiassingers, I've got 10 test machines on various OSes waiting to test this ;)

Feature branch looking good; I have no idea how I totally missed that the tasks was just a simple array... I really should stop tying to program when half asleep...

Using process.platform instead of os.platform shouldn't matter at all in this case, and it saves an extra require.

Thank you for your time to deal with this.

**Comment by matiassingers** [comment]
Haha, I totally know the feeling off programming when not a 100% fresh - suddenly simple solutions end up being super complicated :wink: 

I'll merge the PR then :+1:

**Comment by matiassingers** [comment]
Published in [`1.4.0`](https://github.com/matiassingers/ipa-metadata/releases/tag/1.4.0)
